frontend/src/metabase/dashboard/components/DashboardTabs/use-sync-url-slug.ts

```diff
@@ -66,6 +66,26 @@ export function useSyncURLSlug({ location }: { location: Location }) {
   const { updateURLSlug } = useUpdateURLSlug({ location });
 
   useEffect(() => {
+    if (!tabs || tabs.length === 0) {
+      return;
+    }
+
+    const maybeSelectedTab = tabs.find(t => t.id === getIdFromSlug(slug));
+    if (
+      !prevTabs?.length &&
+      maybeSelectedTab &&
+      selectedTabId !== maybeSelectedTab.id
+    ) {
+      dispatch(selectTab({ tabId: maybeSelectedTab.id }));
+      updateURLSlug({
+        slug: getSlug({
+          tabId: maybeSelectedTab.id,
+          name: maybeSelectedTab.name,
+        }),
+      });
+      return;
+    }
```

Comment:
Could you please explain what case are we handling here?

Reply:
I added this because there was a case that seemed to be missing from the tab logic - the one where we are loading a dashboard with a tab slug, but the tab data hasn't loaded yet. Before the addition of this code, this hook would take the slug and go to line 89:
```
const slugChanged = slug && slug !== prevSlug;
```
Since `prevSlug` is `undefined` (because the dashboard has just loaded), it would say that the slug has changed, and call `initTabs`. `initTabs` then takes this slug, and compares it against the previous dashboard/tabs:
```
const { prevTabs } = getPrevDashAndTabs({ state });
```

but since the dashboard has just loaded, there is no previous set of tabs to compare against. This causes the `state.selectedTabId = null`.

Now, the rest of the code in the useEffect runs smoothly. But now the dashboard loads, and we're calling `getSelectedTabId` which relies on `dashboard`. 

`
`

Reply:
Appreciate a thorough explanation 🙏 I'm a bit surprised this didn't pop up before, we're not rendering `DashboardTabs` component until tabs are loaded, right? Could this be happening because we're now using `useSyncURLSlug` twice (in `SyncedDashboardTabs` and in `DashboardControls`)? I'm not sure why we need the duplication in the first place

Reply:
We never actually use `SyncedDashboardTabs` within a Dashboard component now - all syncing should happen within `DashboardControls` for dashboard components. I did this (and the whole PR) to make sure we don't have any `location` or `replace` calls within the public dashboard, since they mess with client URLs when using the dashboard in the SDK.

Reply:
But for your specific question, it does have to do with the fact that we're loading the dashboard after we initialize `useSyncURLSlug`. I moved it into DashboardControls so we could encapsulate all of the location logic into one place

Reply:
Ahhh, I see it now, thank you! I think it'd be helpful if we could express this in the code itself, at least to some extent. 
For instance, take the condition and assign it to a variable with a meaningful name and/or add a comment
```

Reply:
@kulyk Thanks for calling this out! I've added some code comments to clarify this 👍🏻